optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/service/demo/RoutingProblemList.java

```diff
@@ -33,8 +33,8 @@
 
     private final Map<String, RoutingProblem> routingProblems;
 
-    RoutingProblemList(List<RoutingProblem> routingProblems) {
-        this.routingProblems = Objects.requireNonNull(routingProblems).stream()
+    RoutingProblemList(Stream<RoutingProblem> routingProblems) {
+        this.routingProblems = Objects.requireNonNull(routingProblems)
                 // TODO use file name as the key (that's more likely to be unique than data set name)
                 .collect(toMap(RoutingProblem::name, identity()));
```

Comment:
I'm also not a fan but what this constructor does is essentially parameter validation and making a defensive copy.

I think it is the responsibility of this class to create the map because the way the map is being built is essential for what this class does. There is the `RoutingProblem byName(String name)` method, raising an expectation that this class is able to fetch problems by their name. By building the index here, inside the class (no matter if it's a constructor or a factory method) it's easy to see and test that the class does what it's supposed to. If I took a map from the outside, I'd have to iterate over it and check all entries.